moisan2011.py

```diff
@@ -137,10 +158,22 @@ def per(u, inverse_dft=True):
     is returned.
 
     This function implements Algorithm 1.
+
+    References
+    ----------
+    .. [1] Moisan, Lionel. "Periodic plus smooth image
+           decomposition." Journal of Mathematical Imaging and
+           Vision 39.2 (2011): 161-179.
+           10.1007/s10851-010-0227-1. hal-00388020v2
     """
     u = np.asarray(u, dtype=np.float64)
 
-    m, n = u.shape
+    if axes == (-2, -1):
+        ut = u
+    else:
```

Comment:
Is this `if` really necessary? It does not cover all cases where the `moveaxis` is not necessary, and skeeping this `moveaxis` when it is not required probably leads to a negligible speed gain.

Reply:
agreed and fixed everywhere